rabbitmq_amqp_python_client/connection.py

```diff
@@ -244,4 +266,42 @@ def consumer(
         consumer = Consumer(
             self._conn, destination, message_handler, stream_filter_options, credit
         )
+        self._consumers.append(consumer)
         return consumer
+
+    def _on_disconnection(self) -> None:
+
+        if self in self._connections:
```

Comment:
Here should be a loop with backoff retry the functions `_update_connection` could fail